### PR TITLE
misc: Improve kernel syscall checks

### PR DESCRIPTION
**Pull request by marysaka**
Fix some stuffs around.

Based on kernel 11.14

Depends on #37

### CODE REVIEW

File: include/kernel/kernel.hpp [hunk 1]
```diff
@@ -104,6 +104,8 @@ class Kernel {
 	std::string getProcessName(u32 pid);
 	const char* resetTypeToString(u32 type);
 
+	int copyStringFromUser(u8 *dst, u32 src, u32 size);
```
Comment (Wunkolo) [inline include/kernel/kernel.hpp:107]: Rather than `int` maybe it's time for `i32`/`i16`/`i8`-types.

Comment (wheremyfoodat) [reply include/kernel/kernel.hpp:107]: Yeah s32 sounds good

File: src/core/kernel/kernel.cpp [hunk 2]
```diff
@@ -162,12 +181,23 @@ void Kernel::getSystemTick() {
...
 
 // Result OutputDebugString(const char* str, s32 size)
-// TODO: Does this actually write an error code in r0 and is the above signature correct?
 void Kernel::outputDebugString() {
 	const u32 pointer = regs[0];
-	const u32 size = regs[1];
+	const s32 size = regs[1];
+
+	if (size < 0 || ~pointer < size) {
+		regs[0] = Result::OS::OutOfRange;
```
Comment (wheremyfoodat) [inline src/core/kernel/kernel.cpp:189]: What does the `~pointer < size` do?

**Comment by marysaka** [comment]
I'm not happy about the current PR, I will try to rethink this and open another PR later.

**Pull request closed without merging**